### ISSUE TITLE
build does not include correct binaries

### ISSUE THREAD

**Issue by magnetophon** [curusarn/resh#71]
When I try to build using NixOS, I get the following in /bin/:
```
cli  collect  config  control  daemon  evaluate  event  inspect  postcollect  sanitize  session-init
```

So no ``resh`` or ``reshctl``.

I'm using the following package (that I wrote, not hindered by any expertise!).

Note: I'm not expecting you to debug my nix pkg for me, but I'm just hoping it's something simple.

```
{ stdenv, buildGoPackage, fetchFromGitHub }:

buildGoPackage rec {
  pname = "resh";
  version = "2.5.10";

  goPackagePath = "github.com/curusarn/resh";

  src = fetchFromGitHub {
    owner = "curusarn";
    repo = pname;
    rev = "v${version}";
    sha256 = "1wqaaa1y43id4byjw43ifx25049r3rsz2vd4mi1zfm4kwhswlq2w";
  };

  goDeps = ./deps.nix;

  meta = with stdenv.lib; {
    description = "Rich Enhanced Shell History - Context-based replacement/enhancement for zsh and bash shell history";
    homepage = "https://github.com/curusarn/resh";
    license = licenses.mit;
    maintainers = [ maintainers.magnetophon ];
    platforms = platforms.unix;
  };
}
```
deps.nix was auto-generated, and contains:
```
# file generated from go.mod using vgo2nix (https://github.com/adisbladis/vgo2nix)
[
  {
    goPackagePath = "github.com/BurntSushi/toml";
    fetch = {
      type = "git";
      url = "https://github.com/BurntSushi/toml";
      rev = "v0.3.1";
      sha256 = "1fjdwwfzyzllgiwydknf1pwjvy49qxfsczqx5gz3y0izs7as99j6";
    };
  }
  {
    goPackagePath = "github.com/armon/consul-api";
    fetch = {
      type = "git";
      url = "https://github.com/armon/consul-api";
      rev = "eb2c6b5be1b6";
      sha256 = "1j6fdr1sg36qy4n4xjl7brq739fpm5npq98cmvklzjc9qrx98nk9";
    };
  }
  {
    goPackagePath = "github.com/awesome-gocui/gocui";
    fetch = {
      type = "git";
      url = "https://github.com/awesome-gocui/gocui";
      rev = "v0.6.0";
      sha256 = "19vrxz925i09jhswpsill6556g0v4dnmn7fhjlwizfq05c1vd83w";
    };
  }
  {
    goPackagePath = "github.com/awesome-gocui/termbox-go";
    fetch = {
      type = "git";
      url = "https://github.com/awesome-gocui/termbox-go";
      rev = "c0aef3d18bcc";
      sha256 = "1dydyjdc051y7wgqshaj7y2sgziqz8jxg83a3a6y8xfkvvj6xyq9";
    };
  }
  {
    goPackagePath = "github.com/coreos/etcd";
    fetch = {
      type = "git";
      url = "https://github.com/coreos/etcd";
      rev = "v3.3.10";
      sha256 = "1x2ii1hj8jraba8rbxz6dmc03y3sjxdnzipdvg6fywnlq1f3l3wl";
    };
  }
  {
    goPackagePath = "github.com/coreos/go-etcd";
    fetch = {
      type = "git";
      url = "https://github.com/coreos/go-etcd";
      rev = "v2.0.0";
      sha256 = "1xb34hzaa1lkbq5vkzy9vcz6gqwj7hp6cdbvyack2bf28dwn33jj";
    };
  }
  {
    goPackagePath = "github.com/coreos/go-semver";
    fetch = {
      type = "git";
      url = "https://github.com/coreos/go-semver";
      rev = "v0.2.0";
      sha256 = "1gghi5bnqj50hfxhqc1cxmynqmh2yk9ii7ab9gsm75y5cp94ymk0";
    };
  }
  {
    goPackagePath = "github.com/cpuguy83/go-md2man";
    fetch = {
      type = "git";
      url = "https://github.com/cpuguy83/go-md2man";
      rev = "v1.0.10";
      sha256 = "1bqkf2bvy1dns9zd24k81mh2p1zxsx2nhq5cj8dz2vgkv1xkh60i";
    };
  }
  {
    goPackagePath = "github.com/davecgh/go-spew";
    fetch = {
      type = "git";
      url = "https://github.com/davecgh/go-spew";
      rev = "v1.1.1";
      sha256 = "0hka6hmyvp701adzag2g26cxdj47g21x6jz4sc6jjz1mn59d474y";
    };
  }
  {
    goPackagePath = "github.com/fsnotify/fsnotify";
    fetch = {
      type = "git";
      url = "https://github.com/fsnotify/fsnotify";
      rev = "v1.4.7";
      sha256 = "07va9crci0ijlivbb7q57d2rz9h27zgn2fsm60spjsqpdbvyrx4g";
    };
  }
  {
    goPackagePath = "github.com/go-errors/errors";
    fetch = {
      type = "git";
      url = "https://github.com/go-errors/errors";
      rev = "v1.0.1";
      sha256 = "0rznpknk19rxkr7li6dqs52c26pjazp69lh493l4ny4sxn5922lp";
    };
  }
  {
    goPackagePath = "github.com/hashicorp/hcl";
    fetch = {
      type = "git";
      url = "https://github.com/hashicorp/hcl";
      rev = "v1.0.0";
      sha256 = "0q6ml0qqs0yil76mpn4mdx4lp94id8vbv575qm60jzl1ijcl5i66";
    };
  }
  {
    goPackagePath = "github.com/inconshreveable/mousetrap";
    fetch = {
      type = "git";
      url = "https://github.com/inconshreveable/mousetrap";
      rev = "v1.0.0";
      sha256 = "1mn0kg48xkd74brf48qf5hzp0bc6g8cf5a77w895rl3qnlpfw152";
    };
  }
  {
    goPackagePath = "github.com/jpillora/longestcommon";
    fetch = {
      type = "git";
      url = "https://github.com/jpillora/longestcommon";
      rev = "adb9d91ee629";
      sha256 = "017mbsqg6s8p1f7yv4p95g5qrxrm8fqpb50qgcn5r6kagbf7rbfk";
    };
  }
  {
    goPackagePath = "github.com/magiconair/properties";
    fetch = {
      type = "git";
      url = "https://github.com/magiconair/properties";
      rev = "v1.8.0";
      sha256 = "1a10362wv8a8qwb818wygn2z48lgzch940hvpv81hv8gc747ajxn";
    };
  }
  {
    goPackagePath = "github.com/mattn/go-runewidth";
    fetch = {
      type = "git";
      url = "https://github.com/mattn/go-runewidth";
      rev = "v0.0.8";
      sha256 = "14ilkbhnhl8yj443dd1ga4biapswv4g0b4vm2mix78a8srdww39j";
    };
  }
  {
    goPackagePath = "github.com/mattn/go-shellwords";
    fetch = {
      type = "git";
      url = "https://github.com/mattn/go-shellwords";
      rev = "v1.0.6";
      sha256 = "0s6rcswsr6k5dwcng4ck6xhz5xw6j20x1d24ngbbh9qfxxnw0j3j";
    };
  }
  {
    goPackagePath = "github.com/mb-14/gomarkov";
    fetch = {
      type = "git";
      url = "https://github.com/mb-14/gomarkov";
      rev = "044dd0dcb5e7";
      sha256 = "1z833qhg62hgrjl81s6w4m2pympm3873msa3mm262wg705k3acrl";
    };
  }
  {
    goPackagePath = "github.com/mitchellh/go-homedir";
    fetch = {
      type = "git";
      url = "https://github.com/mitchellh/go-homedir";
      rev = "v1.1.0";
      sha256 = "0ydzkipf28hwj2bfxqmwlww47khyk6d152xax4bnyh60f4lq3nx1";
    };
  }
  {
    goPackagePath = "github.com/mitchellh/go-ps";
    fetch = {
      type = "git";
      url = "https://github.com/mitchellh/go-ps";
      rev = "621e5597135b";
      sha256 = "0sjkz24kqz5nzgncnvdlzhlyba4jwmf6cxj8c4p8mb44679gkphh";
    };
  }
  {
    goPackagePath = "github.com/mitchellh/mapstructure";
    fetch = {
      type = "git";
      url = "https://github.com/mitchellh/mapstructure";
      rev = "v1.1.2";
      sha256 = "03bpv28jz9zhn4947saqwi328ydj7f6g6pf1m2d4m5zdh5jlfkrr";
    };
  }
  {
    goPackagePath = "github.com/pelletier/go-toml";
    fetch = {
      type = "git";
      url = "https://github.com/pelletier/go-toml";
      rev = "v1.2.0";
      sha256 = "1fjzpcjng60mc3a4b2ql5a00d5gah84wj740dabv9kq67mpg8fxy";
    };
  }
  {
    goPackagePath = "github.com/pmezard/go-difflib";
    fetch = {
      type = "git";
      url = "https://github.com/pmezard/go-difflib";
      rev = "v1.0.0";
      sha256 = "0c1cn55m4rypmscgf0rrb88pn58j3ysvc2d0432dp3c6fqg6cnzw";
    };
  }
  {
    goPackagePath = "github.com/russross/blackfriday";
    fetch = {
      type = "git";
      url = "https://github.com/russross/blackfriday";
      rev = "v1.5.2";
      sha256 = "0jzbfzcywqcrnym4gxlz6nphmm1grg6wsl4f0r9x384rn83wkj7c";
    };
  }
  {
    goPackagePath = "github.com/schollz/progressbar";
    fetch = {
      type = "git";
      url = "https://github.com/schollz/progressbar";
      rev = "v1.0.0";
      sha256 = "01qh857wlhghp15s1jh85kp48s9x96jnyk1755rjg3j6nnirhxq8";
    };
  }
  {
    goPackagePath = "github.com/spf13/afero";
    fetch = {
      type = "git";
      url = "https://github.com/spf13/afero";
      rev = "v1.1.2";
      sha256 = "0miv4faf5ihjfifb1zv6aia6f6ik7h1s4954kcb8n6ixzhx9ck6k";
    };
  }
  {
    goPackagePath = "github.com/spf13/cast";
    fetch = {
      type = "git";
      url = "https://github.com/spf13/cast";
      rev = "v1.3.0";
      sha256 = "0xq1ffqj8y8h7dcnm0m9lfrh0ga7pssnn2c1dnr09chqbpn4bdc5";
    };
  }
  {
    goPackagePath = "github.com/spf13/cobra";
    fetch = {
      type = "git";
      url = "https://github.com/spf13/cobra";
      rev = "v0.0.5";
      sha256 = "0z4x8js65mhwg1gf6sa865pdxfgn45c3av9xlcc1l3xjvcnx32v2";
    };
  }
  {
    goPackagePath = "github.com/spf13/jwalterweatherman";
    fetch = {
      type = "git";
      url = "https://github.com/spf13/jwalterweatherman";
      rev = "v1.0.0";
      sha256 = "093fmmvavv84pv4q84hav7ph3fmrq87bvspjj899q0qsx37yvdr8";
    };
  }
  {
    goPackagePath = "github.com/spf13/pflag";
    fetch = {
      type = "git";
      url = "https://github.com/spf13/pflag";
      rev = "v1.0.3";
      sha256 = "1cj3cjm7d3zk0mf1xdybh0jywkbbw7a6yr3y22x9sis31scprswd";
    };
  }
  {
    goPackagePath = "github.com/spf13/viper";
    fetch = {
      type = "git";
      url = "https://github.com/spf13/viper";
      rev = "v1.3.2";
      sha256 = "1829hvf805kda65l59r17wvid7y0vr390s23zfhf4w7vdb4wp3zh";
    };
  }
  {
    goPackagePath = "github.com/stretchr/testify";
    fetch = {
      type = "git";
      url = "https://github.com/stretchr/testify";
      rev = "v1.2.2";
      sha256 = "0dlszlshlxbmmfxj5hlwgv3r22x0y1af45gn1vd198nvvs3pnvfs";
    };
  }
  {
    goPackagePath = "github.com/ugorji/go";
    fetch = {
      type = "git";
      url = "https://github.com/ugorji/go";
      rev = "d75b2dcb6bc8";
      sha256 = "0di1k35gpq9bp958ywranpbskx2vdwlb38s22vl9rybm3wa5g3ps";
    };
  }
  {
    goPackagePath = "github.com/whilp/git-urls";
    fetch = {
      type = "git";
      url = "https://github.com/whilp/git-urls";
      rev = "31bac0d230fa";
      sha256 = "0x2s64mbk4pkmhkfyjv67k5ch49j56i3ys0cgykv1kpaycy8710h";
    };
  }
  {
    goPackagePath = "github.com/xordataexchange/crypt";
    fetch = {
      type = "git";
      url = "https://github.com/xordataexchange/crypt";
      rev = "b2862e3d0a77";
      sha256 = "04q3856anpzl4gdfgmg7pbp9cx231nkz3ymq2xp27rnmmwhfxr8y";
    };
  }
  {
    goPackagePath = "golang.org/x/crypto";
    fetch = {
      type = "git";
      url = "https://go.googlesource.com/crypto";
      rev = "505ab145d0a9";
      sha256 = "1vbsvcvmjz6c00p5vf8ls533p52fx2y3gy6v4k5qrdlzl4wf0i5s";
    };
  }
  {
    goPackagePath = "golang.org/x/sys";
    fetch = {
      type = "git";
      url = "https://go.googlesource.com/sys";
      rev = "a5c9d58dba9a";
      sha256 = "02qv5i7yps35p7fa81345qz7k8i73gkigj69anwmpw9rhpmzayf9";
    };
  }
  {
    goPackagePath = "golang.org/x/text";
    fetch = {
      type = "git";
      url = "https://go.googlesource.com/text";
      rev = "v0.3.0";
      sha256 = "0r6x6zjzhr8ksqlpiwm5gdd7s209kwk5p4lw54xjvz10cs3qlq19";
    };
  }
  {
    goPackagePath = "gopkg.in/check.v1";
    fetch = {
      type = "git";
      url = "https://gopkg.in/check.v1";
      rev = "20d25e280405";
      sha256 = "0k1m83ji9l1a7ng8a7v40psbymxasmssbrrhpdv2wl4rhs0nc3np";
    };
  }
  {
    goPackagePath = "gopkg.in/yaml.v2";
    fetch = {
      type = "git";
      url = "https://gopkg.in/yaml.v2";
      rev = "v2.2.2";
      sha256 = "01wj12jzsdqlnidpyjssmj0r4yavlqy7dwrg7adqd8dicjc4ncsa";
    };
  }
]
```

**Comment by curusarn** [comment]
So `resh` and `reshctl` are not actually binaries. They are shell functions that wrap the actual binaries. This way both `resh` and `reshctl` can interact with the shell session. 

For example: when you run `reshctl enable ctrl_r_binding` we need to enable binding for control+R inside your current terminal session. This is not something that can be done from an executable.

The actual executable behind `reshctl` is `resh-control` and it does most of the work and then `reshctl` function handles the changes inside the shell session.

The actual executable behind `resh` function is `resh-cli`. This is just a convenient way for me to make it possible to call the tool directly instead of using the ctrl+R binding. 


When I develop the project I build it locally using a Makefile.
`make build` builds the project. During the build, all the binaries get prefixed with `resh-`.  
Makefile: https://github.com/curusarn/resh/blob/master/Makefile 

`make install` builds the project and installs it using this script: https://github.com/curusarn/resh/blob/master/scripts/install.sh#L104

Project relies on being installed into `~/.resh/` at the moment and it won't work when moved or installed elsewhere.

Installation among other things adds a line to `~/.bashrc` and `~/.zshrc`. This line sources `~/.resh/shellrc` which sets up many things to make the project work including `resh` and `reshctl` functions. 

I hope this helps a bit. 

TL;DR
`resh` and `reshctl` are a shell functions that are available when `~/.shellrc` is sourced. It's perfectly fine that they are not between binaries after compiling the project.

**Comment by magnetophon** [comment]
All clear, thanks.
I can probably make it work now.
